tiatoolbox/utils/image.py

```diff
@@ -308,6 +355,12 @@ def safe_padded_read(
         bounds = conv_out_size(np.array(bounds), stride=np.tile(stride, 2))
         padded_bounds = bounds + (padding * np.array([-1, -1, 1, 1]))
         img_size = conv_out_size(img_size, stride=stride)
+
+    # Return without padding if pad_mode is none
+    # TODO: Add test for this!
+    if pad_mode in ["none", None]:
+        return region
+
     # Find how much padding needs to be applied to fill the edge gaps
     # edge_padding = np.abs(padded_bounds - clamped_bounds)
```

Comment:
Do you need this line? This should be flagged by flake8

Reply:
This has been removed now